packages/eslint-plugin/tests/docs.test.ts

```diff
@@ -180,6 +181,32 @@ describe('Validating rule docs', () => {
           }
         });
       }
+
+      test('must include only valid code samples', () => {
+        for (const token of tokens) {
+          if (token.type !== 'code') {
+            continue;
+          }
+
+          const lang = token.lang?.trim();
+          if (!lang?.startsWith('ts')) {
```

Comment:
in case you wanna be stricter, and because I just _love_ regex...

```suggestion
          if (!/^tsx?\b/i.test(lang))) {
```

Reply:
Also do we have any `jsx?` blocks?

Reply:
> [Even Nitier] in case you wanna be stricter, and because I just _love_ regex...

IMO, `startsWith` looks more readable, especially for texts, although they are already generated automatically, which makes them more complex than the usual ones...

Reply:
TBH, I think it's also more performant then regex 😅

Reply:
> Also do we have any `jsx?` blocks?

Good catch, thanks, we don't have any `jsx` blocks, but we had a few `js` blocks (I changed them to `ts`)